app/Controllers/LibriController.php

```
@@ -3490,6 +3503,47 @@ private function updateAuthorBioIfEmpty(\mysqli $db, int $authorId, string $bio)
 
+    /**
+     * Resolve the multi-publisher selection (issue #143) into an ordered,
+     * deduplicated list of publisher ids. Combines existing ids (editori_ids[])
+     * with newly typed names (editori_new[], find-or-create), and falls back to
+     * the single primary publisher already resolved from the legacy
+     * editore_search field or from scraping.
+     *
+     * @param array<string,mixed> $data
+     * @return list<int>
+     */
+    private function resolvePublisherIds(\mysqli $db, array $data, ?int $primaryEditoreId): array
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $db. Configured minimum length is 3. (undefined)

(ShortVariable)